cubedash/index/postgis/_api.py

```diff
@@ -0,0 +1,977 @@
+        with self.index._active_connection() as conn:
+            return conn.execute(
+                text(f"""
+            with srid_groups as (
+                select cubedash.dataset_spatial.product_ref                         as product_ref,
+                        cubedash.dataset_spatial.region_code                             as region_code,
+                        ST_Transform(ST_Union(cubedash.dataset_spatial.footprint), 4326) as footprint,
+                        count(*)                                                         as count
+                from cubedash.dataset_spatial
+                where cubedash.dataset_spatial.product_ref = {product_id}
+                        and
+                        st_isvalid(cubedash.dataset_spatial.footprint)
+                group by cubedash.dataset_spatial.product_ref,
+                        cubedash.dataset_spatial.region_code,
+                        st_srid(cubedash.dataset_spatial.footprint)
+            )
+            insert into cubedash.region (product_ref, region_code, footprint, count)
+                select srid_groups.product_ref,
+                    coalesce(srid_groups.region_code, '')                          as region_code,
+                    ST_SimplifyPreserveTopology(
+                            ST_Union(ST_Buffer(srid_groups.footprint, 0)), 0.0001) as footprint,
+                    sum(srid_groups.count)                                         as count
+                from srid_groups
+                group by srid_groups.product_ref, srid_groups.region_code
+            on conflict (product_ref, region_code)
+                do update set count           = excluded.count,
+                            generation_time = now(),
+                            footprint       = excluded.footprint
+            returning product_ref, region_code, footprint, count
+
+                """)
+            )
+
+    def delete_product_empty_regions(self, product_id: int):
+        with self.index._active_connection() as conn:
+            return conn.execute(
+                text(f"""
+            delete from cubedash.region
+            where product_ref = {product_id} and region_code not in (
+                select cubedash.dataset_spatial.region_code
+                from cubedash.dataset_spatial
+                where cubedash.dataset_spatial.product_ref = {product_id}
+                group by cubedash.dataset_spatial.region_code
+            )
+                """),
+            )
+
+    def product_time_overview(self, product_id: int):
+        with self.index._active_connection() as conn:
+            return conn.execute(
+                select(
+                    func.min(DatasetSpatial.center_time),
+                    func.max(DatasetSpatial.center_time),
+                    func.count(),
+                ).where(DatasetSpatial.product_ref == product_id)
+            ).fetchone()
+            # return conn.execute(
```

Comment:
Do we need this commented out code?

Reply:
Nope, there were just so many of them scattered around that I've missed a few!